kalman_filter.py

```diff
@@ -39,60 +39,66 @@ def update(self, z):
         
     
-    # TODO Part 3: Implement here the measurement model
+    # Part 3: Implement here the measurement model
     def measurement_model(self):
-        x, y, th, w, v, vdot = self.x
+        # Don't need x, y, or th when unpacking
+        _, _, _, w, v, vdot = self.x
+
         return np.array([
-            ...,# v
-            ...,# w
-            ..., # ax
-            ..., # ay
+            v, # v - linear velocity along the forward (x) direction
+            w, # w - angular velocity
+            vdot, # ax - acceleration in the forward direction
+            v * w, # ay - lateral acceleration
         ])
         
-    # TODO Part 3: Impelment the motion model (state-transition matrice)
+    # Part 3: Implement the motion model (state-transition matrix)
     def motion_model(self):
-        
         x, y, th, w, v, vdot = self.x
         dt = self.dt
         
+        # vdot and w are constant over the timestep
         self.x = np.array([
-            x + ... * np.cos(th) * dt,
-            y + ... * np.sin(th) * dt,
+            x + v * np.cos(th) * dt,
+            y + v * np.sin(th) * dt,
             th + w * dt,
             w,
-            v  + vdot*dt,
-            vdot,
+            v + vdot*dt,
+            vdot
         ])
-        
-
 
-    
     def jacobian_A(self):
-        x, y, th, w, v, vdot = self.x
+        # Don't need x, y, w, vdot when unpacking
+        _, _, th, _, v, _ = self.x
         dt = self.dt
         
+        # 
         return np.array([
-            #x, y,               th, w,             v, vdot
-            [1, 0,              ..., 0,          ...,  0],
-            [0, 1,              ..., 0,          ...,  0],
+            #x, y,               th, w,               v, vdot
+            [1, 0,              -v * np.sin(th) * dt, 0, np.cos(th) * dt,  0],
+            [0, 1,              v * np.cos(th) * dt, 0,  np.sin(th) * dt,  0],
             [0, 0,                1, dt,           0,  0],
             [0, 0,                0, 1,            0,  0],
             [0, 0,                0, 0,            1,  dt],
             [0, 0,                0, 0,            0,  1 ]
         ])
     
     
-    # TODO Part 3: Implement here the jacobian of the H matrix (measurements)    
+    # Part 3: Implement here the jacobian of the H matrix (measurements)    
     def jacobian_H(self):
-        x, y, th, w, v, vdot=self.x
+        # Dont need x, y, th, or vdot when unpacking
+        _, _, _, w, v, _ = self.x
+
+        # partial(ay) with respect to w is v
+        # partial(ay) with respect to v is w
         return np.array([
             #x, y,th, w, v,vdot
             [0,0,0  , 0, 1, 0], # v
             [0,0,0  , 1, 0, 0], # w
             [0,0,0  , 0, 0, 1], # ax
-            [0,0,0  , ..., ..., 0], # ay
+            [0,0,0  , v, w, 0], # ay
         ])
         
-    # TODO Part 3: return the states here    
+    # Part 3: return the states here    
     def get_states(self):
```

Comment:
Can modify this as needed

Reply:
How did you get ay= v * w in Line 51? Aside from that and Line 98 which derives from that everything else looks good